python/src/DiscreteMarkovChain_doc.i.in

```diff
@@ -0,0 +1,89 @@
+%feature("docstring") OT::DiscreteMarkovChain
+"Discrete Markov chain process.
+
+Parameters
+----------
+origin : int, :math:`origin \geq 0`
+    Origin of the Markov chain, i.e. state of the process at :math:`t_0`.
+
+transitionMatrix : :class:`~openturns.Matrix` 
+    Transition matrix of the process. The matrix must be square, of dimension equal
+    to the number of possible states of the process.
+
+timeGrid : :class:`~openturns.TimeSeries`, optional
+    The time grid of the process.
+    By default, the time grid is  reduced to one time stamp equal to 0.
+
+
+Notes
+-----
+A discrete Markov chain is a process :math:`X: \Omega \times \cD \rightarrow E`, where :math:`\cD=\Rset` discretized on the time grid :math:`(t_i)_{i \geq 0}`, and :math:`E = [\![ 1,...,n]\!]` is the space of states, such that:
+
+.. math::
+
+    \forall n>0,\: \Pset ( X_{t_n} | X_{t_0},...X_{t_{n-1}} )  =  \Pset ( X_{t_n} | X_{t_{n-1}} )
+    
+The transition matrix of the process :math:`\cM = (m_{i,j})` can be defined such that:
+
+.. math::
+
+     \forall t_n \in \cD, \forall i,j \leq n, m_{i,j} = \Pset (X_{t_{n+1}} = j | X_{t_{n}} = i)
+
+The transition matrix :math:`\cM` of the process is square, and its dimension :math:`n` is equal to the number of states of the process. Besides, :math:`\cM` is a stochastic matrix, i.e.:
```

Comment:
If you reverse the role of i and j you get a structure much more adapted to the memory storage of Matrix